src/install/lockfile/Tree.zig

```diff
@@ -534,9 +551,48 @@ pub fn processSubtree(
             }
         }
 
-        const dependency = builder.dependencies[dep_id];
-
         const hoisted: HoistDependencyResult = hoisted: {
+            if (comptime method == .filter) {
+                // not filtered, but does it match a nohoist pattern?
+                if (builder.manager.nohoist_patterns.len != 0) try_nohoist: {
+                    const string_buf = builder.lockfile.buffers.string_bytes.items;
+
+                    try dep_subpath.ensureTotalCapacity(subpath.items().len + @intFromBool(subpath.items().len != 0) + dependency.name.len());
+                    dep_subpath.appendSliceAssumeCapacity(subpath.items());
+                    if (subpath.items().len != 0) {
+                        dep_subpath.appendAssumeCapacity('/');
+                    }
+                    dep_subpath.appendSliceAssumeCapacity(dependency.name.slice(string_buf));
```

Comment:
This feels janky and kind of hard to read. You repeat yourself in `subpath.items().len != 0`. Why not:

```zig
if (subpath.items().len == 0) {
    try dep_subpath.append('/');
} else {
    try dep_subpath.ensureTotalCapacity(subpath.items().len + dependency.name.len());
    try dep_subpath.appendSliceAssumeCapacity(subpath.items());
    const string_buf = builder.lockfile.buffers.string_bytes.items;
    try dep_subpath.appendSliceAssumeCapacity(dependency.name.slice(string_buf));
}
```

This is a lot more digestible since I don't have to keep track of the condition in my head twice.